src/geometry.cpp

```diff
@@ -79,4 +81,32 @@ namespace geometry {
 	  return v;
 
 	}
+
+	std::vector<Cell> bresenham_conversion(const Cell &start, const Cell &end){
+
+		const auto pixels = bresenham_conversion(start.x, start.y, end.x, end.y);
+
+		std::vector<Cell> pixels_return;
+
+		for (const auto &pixel:pixels) {
+			Cell temp {pixel[0],pixel[1]};
+			pixels_return.push_back(temp);
+		}
+
+		return pixels_return;
+
+	}
+
+	std::vector<Cell> raytrace(const Cell& start, const Cell& end, int max_length) {
+	  if (max_length < 1) {
+	  	return std::vector<Cell>();
+	  }
+	  // Get the interpolated pixels bresenham
+	  const auto pixels = bresenham_conversion(start, end);	 
+
+	  return {pixels.begin(), (pixels.begin() + std::min(static_cast<int>(pixels.size()),max_length))};
```

Comment:
you don't need the extra parens ~(~`pixels.begin()`

Reply:
Done